text/0039-burn-redemption-memo.md

```diff
@@ -0,0 +1,75 @@
+
+
+# Motivation
+[motivation]: #motivation
+
+With the upcoming addition of minting [MCIP #0037](https://github.com/mobilecoinfoundation/mcips/pull/37/) and the already-existing [MCIP #0035](0035-verifiable-burning.md) the path for adding and removing tokens from circulation on the MobileCoin blockchain has been established.
+It is expected that burning (the action of permanently removing tokens from circulation on the MobileCoin blockchain) would often be correlated with an event happening outside of the MobileCoin blockchain. For example, it is possible for a bridge between the MobileCoin blockchain and some other external blockchain to exist, that either manually or automatically:
+* Mints new tokens on the MobileCoin blockchain when handed tokens from the external blockchain.
+* Burns (see [MCIP #0035](0035-verifiable-burning.md)) tokens on the MobileCoin blockchain and hands out tokens on the external blockchain.
+
+When burn operations of this nature take place, it is desirable to have the option of including data in the MobileCoin transaction that somehow ties it to the external event that triggerred the burn. This will allow such bridge operators and outside auditors scanning the blockchain for burn transactions to tie the burn to the operation that happens outside of the MobileCoin blockchain.
+
+Recoverable Transaction History ([MCIP #4](0004-recoverable-transaction-history.md)) provides a way to attach up to 64 bytes of memo to a `TxOut`. Additionally, it contains two bytes for specifying the memo type, so that clients know how to parse the memo contents. We propose adding a new memo type that is used for indicating burn `TxOut`s that contain 64-bytes of unstructured payload. The reason the payload is unstructured is that every token on the MobileCoin network might (and will likely) have different burn semantics. Additionally, this data is used by the entity performing the burn to do their own bookkeeping, and it is unnecessary for us to put constraints on what they can store. This is acceptable since this memo is not meant to be read by regular MobileCoin wallet implementations (since most wallets will not be view-key-scanning the burn address at all).
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+
+This is a simple proposal. We propose reserving the memo type 0x0001 for the burn redemption memo:
+
+| Memo type bytes | Name                                              |
+| -----------     | -----------                                       |
+| 0x0001          | Burn redemption memo                              |
+
+
+## 0x0001 Burn redemption memo
+
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 64     | Unstructured payload  |
```

Comment:
This all seems perfectly logical to make completely unstructured so that we're not making assumptions. If we need something more, we can add something more.

However, does it make sense for us to possible reserve memo types that might fall under the classification of "burn tx" (similar to how 0x0200 are change memos & 0x0100 are destination memos) that can be diversified under the superclass of "burn memos"? It could feasibly in the future need more structure due to excessive fragmentation or regulation pressure. Say we need a "chain identifier + burn tx indentifier" or something, is there an address space of memo types we can currently reserve for this (say the space under 0x010)?

Reply:
Gift code memos were reserved at 0x0002 and it's feasible other memo types could just start popping up behind that. We could just implement 0x0003.. 0x0004.. etc. - but maybe given this is very regulation heavy, we create a class for it if we need more space?

Reply:
I don't think there's a requirement for the memo type bytes to be sequential, so leaving gaps doesn't seem necessary to me but I also don't have any strong preference about that so we could move this to be 0x0010 and leave a few unused ones before it.

Reply:
Yeah there's definitely no strict requirement for it. Let's see what other people think

Reply:
This MCIP has my approval either way. If other people think we should reserve some address space, that's fine, otherwise I think we're good to reserve space only as needed